ci/run_test/main.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# This is the script used for the run_tests.sh stage in CI
+# for doing unit testing.
+set -ex
+
+# JENKINS-52781 tar function is breaking symlinks
+rm -rf test_results
+mkdir test_results
+# shellcheck disable=SC1091
+. ./.build_vars.sh
+rm -f "${SL_BUILD_DIR}/src/control/src/github.com/daos-stack/daos/src/control"
+mkdir -p "${SL_BUILD_DIR}/src/control/src/github.com/daos-stack/daos/src/"
+ln -s ../../../../../../../../src/control \
+  "${SL_BUILD_DIR}/src/control/src/github.com/daos-stack/daos/src/control"
+DAOS_BASE=${SL_PREFIX%/install*}
+rm -f dnt.*.memcheck.xml vm_test.out nlt-errors.out
+NODE=${NODELIST%%,*}
+mydir="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+
+# shellcheck disable=SC2029
+ssh "$SSH_KEY_ARGS" jenkins@"$NODE" "DAOS_BASE=$DAOS_BASE      \
+                                     HOSTNAME=$HOSTNAME        \
+                                     PWD=$PWD                  \
+                                     SL_PREFIX=$SL_PREFIX      \
+                                     $(cat "$mydir/main_node.sh")"
```

Comment:
I would much rather this did a mount, and then invoked the main_node.sh script via nfs.  Escaping the contents of main_node.sh like this makes it much harder to write/debug things in that script.

Reply:
We want to remove mounting the Jenkins workspace on the CI nodes long term.

Will implement doing an scp on the file instead of doing the cat on the contents in the next push.

Reply:
> I would much rather this did a mount, and then invoked the main_node.sh script via nfs. Escaping the contents of main_node.sh like this makes it much harder to write/debug things in that script.

But the scriptlets in the files don't need to be escaped.  `cat`ting in an `ssh` command eliminates two remote operations, reducing "environmental" risk.

Reply:
Changing it to copy the file instead of using cat to insert it found an issue with using PWD to pass the host $PWD to the remote system.

Can you and Ashley work out which way you want this?

Reply:
Changed back to use cat instead of scp.